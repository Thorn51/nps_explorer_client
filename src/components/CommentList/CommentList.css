/* Colors */
:root {
  --darkGunmetal: #122932;
  --blackCoral: #576066;
  --queenBlue: #386fa4;
  --middleBlue: #80ced7;
  --mintCream: #f7fff7;
  --platinum: #e0e2e3;
}

/* User Feedback */
.user_feedback {
  display: flex;
  flex-direction: column;
  margin-top: 25px;
  width: 90%;
  margin-left: auto;
  margin-right: auto;
}

.comment {
  border: 2px solid var(--darkGunmetal);
  margin: 10px 0 10px 0;
  box-shadow: 2px 2px 2px black;
}

.user {
  width: 100%;
  color: var(--platinum);
  padding: 10px;
  background-color: var(--darkGunmetal);
}

.comment_text {
  padding: 10px;
}

.comment_form {
  width: 100%;
}
