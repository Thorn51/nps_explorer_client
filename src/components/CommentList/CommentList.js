import React, { useContext, useState, useEffect } from "react";
import { GlobalContext } from "../../context/GlobalState";
import Comment from "../Comment/Comment";
import CommentForm from "../../components/CommentForm/CommentForm";
import "./CommentList.css";

export default function CommentList(props) {
  const { comments, getComments } = useContext(GlobalContext);

  let [isBusy, setBusy] = useState(true);

  // Get comments from database
  useEffect(() => {
    getComments();
    setBusy(false);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const renderParkComments = () => {
    let parkComments = comments.filter(
      comment => comment.parkCode === props.parkCode
    );
    if (parkComments.length > 0) {
      return (
        <section className="feedback">
          <h3 className="section_title">Comments</h3>
          {parkComments.map(comment => (
            <Comment
              key={comment.id}
              author={comment.authorName}
              date={comment.dateSubmitted}
              comment={comment.commentText}
            />
          ))}
          <CommentForm parkCode={props.parkCode} parkName={props.parkName} />
        </section>
      );
    } else {
      return (
        <section className="feedback">
          <h3 className="section_title">Comments</h3>
          <p>Be the first to comment.</p>
          <CommentForm parkCode={props.parkCode} parkName={props.parkName} />
        </section>
      );
    }
  };

  return (
    <>
      {isBusy === false ? (
        renderParkComments()
      ) : (
        <section className="feedback">
          <h3 className="section_title">Gathering Comments</h3>
        </section>
      )}
    </>
  );
}
