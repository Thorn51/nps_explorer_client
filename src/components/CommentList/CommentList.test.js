import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import CommentList from "./CommentList";

describe("<CommentList />", () => {
  it("Renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<CommentList />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("Renders the UI as expected", () => {
    const tree = renderer.create(<CommentList />).toJSON();
    expect("tree").toMatchSnapshot();
  });
});
