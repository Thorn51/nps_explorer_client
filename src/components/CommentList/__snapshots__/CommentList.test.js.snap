// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentList /> Renders the UI as expected 1`] = `"tree"`;
